src/reconcile/rec_row.c

```diff
@@ -812,18 +812,12 @@ __wt_rec_row_leaf(
                 dictionary = true;
             } else if (vpack->raw == WT_CELL_VALUE_OVFL_RM) {
                 /*
-                 * If doing an update save and restore, and the underlying value is a removed
-                 * overflow value, we end up here.
+                 * If the underlying value is a removed overflow value, we end up here.
                  *
                  * If necessary, when the overflow value was originally removed, reconciliation
                  * appended a globally visible copy of the value to the key's update list, meaning
                  * the on-page item isn't accessed after page re-instantiation.
                  *
-                 * Assert the case.
-                 */
-                WT_ASSERT(session, F_ISSET(r, WT_REC_UPDATE_RESTORE));
-
-                /*
```

Comment:
Presumably this whole chunk of code should be removed, not just the assert: can this happen apart from a `WT_REC_UPDATE_RESTORE` configuration?